xbmc/filesystem/FileDirectoryFactory.cpp

```diff
@@ -64,20 +64,22 @@ IFileDirectory* CFileDirectoryFactory::Create(const CURL& url, CFileItem* pItem,
 
   std::string strExtension=URIUtils::GetExtension(url);
   StringUtils::ToLower(strExtension);
-  VECADDONS codecs;
-  CAddonMgr::GetInstance().GetAddons(codecs, ADDON_AUDIODECODER);
-  for (size_t i=0;i<codecs.size();++i)
+  if (!strExtension.empty())
```

Comment:
Every call to GetAddons() needs > 20 ms time, if this is done repeatadly, you will kill the mainloop. So if that will be the case with the other PR (in some combination) you will endup with issues on every platform.

So then pick this commit and revert it if this is your final say.